src/main/java/io/stargate/sgv2/jsonapi/service/embedding/operation/MeteredEmbeddingProvider.java

```diff
@@ -0,0 +1,74 @@
+package io.stargate.sgv2.jsonapi.service.embedding.operation;
+
+import io.micrometer.core.instrument.*;
+import io.smallrye.mutiny.Uni;
+import io.stargate.sgv2.api.common.config.MetricsConfig;
+import io.stargate.sgv2.jsonapi.api.request.DataApiRequestInfo;
+import io.stargate.sgv2.jsonapi.api.v1.metrics.JsonApiMetricsConfig;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import java.util.List;
+import java.util.Optional;
+
+@ApplicationScoped
+public class MeteredEmbeddingProvider implements EmbeddingProvider {
+  MeterRegistry meterRegistry;
+  JsonApiMetricsConfig jsonApiMetricsConfig;
+  DataApiRequestInfo dataApiRequestInfo;
+  private static final String UNKNOWN_VALUE = "unknown";
+  private final MetricsConfig.TenantRequestCounterConfig tenantConfig;
+  EmbeddingProvider embeddingClient;
+
+  String commandName;
+
+  @Inject
+  public MeteredEmbeddingProvider(
+      MeterRegistry meterRegistry,
+      JsonApiMetricsConfig jsonApiMetricsConfig,
+      DataApiRequestInfo dataApiRequestInfo,
+      MetricsConfig metricsConfig) {
+    this.meterRegistry = meterRegistry;
+    this.jsonApiMetricsConfig = jsonApiMetricsConfig;
+    this.dataApiRequestInfo = dataApiRequestInfo;
+    tenantConfig = metricsConfig.tenantRequestCounter();
+  }
+
+  public MeteredEmbeddingProvider setEmbeddingClient(
```

Comment:
This access pattern is incorrect. You are setting embeddingClient on a ApplicationScoped object. The class needs to be @RequestScoped with providers for every fields or create a regular java object.

Reply:
Changed to `@RequestScoped`

Reply:
Change to a regular java object